iphone/Maps/Bookmarks/BookmarksList/BookmarksListInterfaces.swift

```diff
@@ -51,7 +56,7 @@ protocol IBookmarksListView: AnyObject {
   func setInfo(_ info: IBookmarksListInfoViewModel)
   func setSections(_ sections: [IBookmarksListSectionViewModel])
   func setMoreItemTitle(_ itemTitle: String)
-  func showMenu(_ items: [IBookmarksListMenuItem])
+  func showMenu(_ items: [IBookmarksListMenuItem], from source: BookmarkToolbarButtonSource)
```

Comment:
Thanks! I like this approach and how it works!
But what do you think: is it worth to pass this `BookmarkToolbarButtonSource` through the all cycle from view->presenter->vm->view? 

For example replace in presenter methods (because the pressing the sort/more button means not actually `make the sort/more`, but `show the _source_ menu`):
```swift
  func sort()
  func more()
```
with (like in VM)
```swift
 func showMenu(from source: BookmarkToolbarButtonSource)
```

so we can call in VC:
```swift
  @IBAction private func onSortItem(_ sender: UIBarButtonItem) {
    presenter.showMenu(from: .sort)
  }
  @IBAction private func onMoreItem(_ sender: UIBarButtonItem) {
     presenter.showMenu(from: .more)
  }
```
and pass this source further

Reply:
I think with the current approach its clean and addresses the issue so i'm not sure i see a benefit of changing it.

Reply:
@kirylkaveryn did you have a specific overcomplication in mind with the interactor that can be simplified?

Reply:
> @kirylkaveryn did you have a specific overcomplication in mind with the interactor that can be simplified?

I think that it's clearer to set the `source` when calling, and it's not presenter's responsibility to determine what the `source` should be.

Reply:
Ok, thanks. @v-lozko FYI, historically, the written code may not be the best possible implementation. So sometimes instead of just slightly patching or reusing existing patterns, we prefer to refactor and simplify the code. The general rule is that the less abstract patterns and less code are better.